src/Resources/contao/classes/FrontendTemplate.php

```diff
@@ -414,4 +328,39 @@ public function getCustomSections($strKey=null)
 
 		return '<div class="custom">' . "\n" . $sections . "\n" . '</div>' . "\n";
 	}
+
+	/**
+	 * Set the cache headers according to the page settings.
+	 *
+	 * @param Response $response
+	 */
+	private function setCacheHeaders(Response $response)
+	{
+		/** @var $objPage \PageModel */
+		global $objPage;
+
+		if (false === $objPage->cache && false === $objPage->clientCache)
+		{
+			return;
+		}
+
+		// If FE_USER_LOGGED_IN or BE_USER_LOGGED_IN every request is private
+		// Moreover, mobile layout and protected pages are not cached either
+		// TODO: Add support for proxies so they can vary on member context and page layout
+		if (true === FE_USER_LOGGED_IN || true === BE_USER_LOGGED_IN || true === $objPage->isMobile || $objPage->protected)
+		{
+			$response->setPrivate();
```

Comment:
Disagree. It's more explicit like that. The other way would require one to know that `setSharedMaxAge` does reset the private header.